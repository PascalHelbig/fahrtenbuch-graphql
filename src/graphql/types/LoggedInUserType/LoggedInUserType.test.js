const { makeExecutableSchema, addMockFunctionsToSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const LoggedInUserType = require('./LoggedInUserType');

it('has to be a valid schema', () => {
  const SchemaDefinition = 'schema { query: LoggedInUser }';
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, LoggedInUserType] });
  addMockFunctionsToSchema({
    schema,
    mocks: {
      LoggedInUser: () => ({
        user: () => ({ id: 1337 }),
        email: () => 'email@example.com',
        memberships: () => [
          { group: { id: 1338 } },
          { group: { id: 1339 } },
        ],
        boats: () => [{ id: 1340 }, { id: 1341 }],
        availableBoats: () => [{ id: 1341 }],
        participations: () => [{ id: 1342 }, { id: 1343 }],
      }),
    },
  });

  const query = `
    query {
      email
      user { id }
      memberships { 
        group { id } 
      }
      boats { id }
      availableBoats { id }
      participations { id }
    }
  `;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot(),
  );
});
