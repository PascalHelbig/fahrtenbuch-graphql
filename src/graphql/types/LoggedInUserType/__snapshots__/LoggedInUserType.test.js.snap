exports[`test has to be a valid schema 1`] = `
Object {
  "data": Object {
    "availableBoats": Array [
      Object {
        "id": "1341",
      },
    ],
    "boats": Array [
      Object {
        "id": "1340",
      },
      Object {
        "id": "1341",
      },
    ],
    "email": "email@example.com",
    "memberships": Array [
      Object {
        "group": Object {
          "id": "1338",
        },
      },
      Object {
        "group": Object {
          "id": "1339",
        },
      },
    ],
    "participations": Array [
      Object {
        "id": "1342",
      },
      Object {
        "id": "1343",
      },
    ],
    "user": Object {
      "id": "1337",
    },
  },
}
`;
